lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1963,6 +1963,11 @@ LogicalResult ConvertLayoutOpConversion::lowerBlockedToShared(
   smemBase = bitcast(elemPtrTy, smemBase);
   unsigned numWordsEachRep = product<unsigned>(wordsInEachRep);
   SmallVector<Value> wordVecs(numWordsEachRep);
+  // TODO: We should get less barriers if it is handled by membar pass
+  //       instead of the backend, since the later can only handle it in
+  //       the most conservative way. However just keep for now and revisit
+  //       in the future in case necessary.
+  barrier;
```

Comment:
Could we make this have a function syntax, i.e. `barrier();` ? Without the `()`, this line can be misinterpreted as having no side effect

Reply:
Is your concern that gcc would misinterpret and move its position?  I don't think so since it's just a maco, and it's substantially an opbuilder.create() call, which should be safe. 
Anyway, i'll turn it into barrier() to avoid misunderstanding.

Reply:
done.

Reply:
My concern was not so much for GCC but for human readers :p Until I looked at the macro, I thought that there was a typo in the code. Thanks! Will merge now